gspread/worksheet.py

```diff
@@ -2607,6 +2607,28 @@ def unmerge_cells(self, name):
 
         return self.spreadsheet.batch_update(body)
 
+    def get_notes(self):
+        """Returns a list of lists containing all notes in the sheet, or the empty list if the
+        sheet does not have a note.
+        """
+        url = SPREADSHEET_URL % (self.spreadsheet.id)
+        params = {"fields": "sheets/data/rowData/values/note"}
+        response = self.client.request("get", url, params=params)
+        response.raise_for_status()
+        response_json = response.json()
+
+        try:
+            data = response_json["sheets"][self.index]["data"][0]["rowData"]
+            notes = []
+            for row in data:
+                notes.append([])
+                for cell in row.get("values", []):
+                    notes[-1].append(cell.get("note", ""))
```

Comment:
please explain the reason why you add an empty list into the list `notes`. this creates a matrix (may be you were looking for the method `extend()` ?

I'd rather start with an empty list (as you did so far), and then add the notes to this list.
the `row.get("values", [])` is a good approach. 
Then all you need is: append the note if any, and here you should use:
`notes.append(cell.get("note"))` which will get the note string value if any or None if none exists in the list.
appending None will have no effect, so the list only grows as we append values.

Reply:
Yes. I think a matrix would be better since we can know the row index and column index of specific note.
Also, the function get_values returns a 2d-array. I want get_values and get_notes correspond to each other.
Thanks.

Reply:
I see, that is a good point ! then I'll suggest:

- you set the default value to `None` which is the type used to define an empty value
- you should add a new parameter to the function `default_empty_value` so the user can choose what to use as empty value, some might want/need empty string `""` others might want something else (like `None`).